### PR TITLE
adjust shared actions to work with Node's CI

### PR DESCRIPTION
**Pull request by baileympearson**
Key changes:

- only one of Node's packages generates an augmented SBOM file, so silk_asset_group is now optional and only augments the SBOM lite file if the input is present
- Node doesn't have release branches for minors, so the ref for sarif report generation is not the same as the release version.  There is a new input to specify the ref.
- The 3rd party dependency tracking tool is now an input, and defaults to `n/a` if unspecified in the compliance report document.
- The augmented sbom filename is now a required input to the SBOM action, and an optional input with a default in the full report action.

This PR takes the place of https://github.com/mongodb-labs/drivers-github-tools/pull/38.
Example integration into the Node driver: https://github.com/mongodb/node-mongodb-native/actions/runs/9603785418

### CODE REVIEW

File: full-report/action.yml
Comment (baileympearson) [inline full-report/action.yml:46]: Will revert this once I have approvals, leaving this as a reminder to do so

File: compliance-report/generate.sh
Comment (blink1073) [inline compliance-report/generate.sh:34]: This is backwards incompatible, can we default to Silk instead of n/a?

File: compliance-report/generate.sh
Comment (alcaeus) [inline compliance-report/generate.sh:28]: Nit: Couldn't this be replaced with a `default: Silk` line in the corresponding action.yml (in keeping with Steve's suggestion for BC)?

File: full-report/action.yml
Comment (alcaeus) [inline full-report/action.yml:12]: This would technically require us to tag `v3` - any way we can default this to empty and use the version name as fall back?

File: full-report/action.yml
Comment (alcaeus) [reply full-report/action.yml:50]: Good idea. I did take a crack at fixing #37 and allow using the `github.action_repository` context, but that doesn't work unfortunately :(

File: upload-s3-assets/action.yml
Comment (alcaeus) [inline upload-s3-assets/action.yml:31]: We've generally tried to avoid printing file contents as it may contain information that is not supposed to be public (but not exactly sensitive either). Any opinion @blink1073?

Comment (baileympearson) [reply upload-s3-assets/action.yml:31]: I can revert this, but I found that without printing the contents of the files it was impossible to determine whether or not the action was working properly without uploading the files to S3 (which makes the dry_run logic here irrelevant).  This would be useful to have while I incorporate the action into Node's other repos, but if we don't want to print I can just upload files to s3 to confirm things are working.

File: full-report/action.yml
Comment (baileympearson) [reply full-report/action.yml:12]: I started a conversation on slack about whether breaking changes are acceptable.  But I'll update this regardless

File: compliance-report/generate.sh
Comment (baileympearson) [reply compliance-report/generate.sh]: Done

File: compliance-report/generate.sh
Comment (baileympearson) [reply compliance-report/generate.sh]: Actually, I went with Andreas' suggestion (defaulting it in the action.yml).

File: upload-s3-assets/action.yml
Comment (blink1073) [reply upload-s3-assets/action.yml:31]: I think the only file that is somewhat sensitive currently is the SARIF file, but I agree it would be a bad decision to leave this here for all files.

Comment (blink1073) [reply upload-s3-assets/action.yml:31]: You could instead print something like the stats of the file.

File: full-report/action.yml
Comment (blink1073) [reply full-report/action.yml]: I agree we should keep backwards compat as a matter of practice unless we explicitly want to move to `v3`.

File: sbom/action.yml
Comment (alcaeus) [inline sbom/action.yml:9]:
```suggestion
    default: cyclonedx.sbom.json
```

**Review by alcaeus — APPROVED** [review]
LGTM!

**Review by blink1073 — APPROVED** [review]
LGTM!

**Pull request merged**
